Move tool event functions out of top toolbar

AaronVanGeffen opened the pull request:
This PR moves the terraform tool event functions out of the top toolbar window, reducing the complexity considerably.

- [x] Might also be able to localise a couple of globals as a result of this.
- [x] Fix windowless custom tools defined in user scripts.
- [x] Needs some thorough testing of the terraform and scenery tools.
- [x] Take advantage of reduced complexity to fix 'toolbarless' terraform shortcuts.

karst commented:
![image](https://github.com/user-attachments/assets/419dd9b8-4d13-4e17-aa7a-7996a0030404)
I have not noticed any differences in behaviour

AaronVanGeffen commented:
Looks like custom tools (i.e. those from scripts) may be broken by this, particularly those that are active without a visible window. Maybe `CustomWindow.cpp` isn't the best location for firing their events, then. Still, neither is `TopToolbar.cpp`. Needs to be fixed before this is merged, though, clearly.

AaronVanGeffen commented:
Tried fiddling with it, but moving the script tool events into `MouseInput.cpp` wasn't viable either, nor exactly clean. Particularly problematic is that `ToolCancel` is not used exclusively in the UI subproject, so it doesn't have access to the custom tools from the script engine.

For now, I think it best to leave the custom tool events in the `TopToolbar.cpp` file. I have cleaned up the functions somewhat, though, and added a comment with the rationale for having them here.

Gymnasiast commented on src/openrct2/windows/_legacy.cpp:
```diff
@@ -415,8 +415,10 @@ bool SceneryToolIsActive()
     if (InputTestFlag(INPUT_FLAG_TOOL_ACTIVE))
-        if (toolWindowClassification == WindowClass::TopToolbar && toolWidgetIndex == WC_TOP_TOOLBAR__WIDX_SCENERY)
+    {
+        if (toolWindowClassification == WindowClass::Scenery && toolWidgetIndex == 0) // TODO: global widx
```
TODO still needs addressing.

Gymnasiast reviewed (approved):
Testing revealed no bugs.
